x-pack/legacy/plugins/apm/typings/elasticsearch/aggregations.ts

```diff
@@ -0,0 +1,239 @@
+    interval: number;
+    min_doc_count?: number;
+    extended_bounds?: {
+      min: number;
+      max: number;
+    };
+  };
+  avg: MetricsAggregationOptions;
+  max: MetricsAggregationOptions;
+  min: MetricsAggregationOptions;
+  sum: MetricsAggregationOptions;
+  value_count: MetricsAggregationOptions;
+  cardinality: MetricsAggregationOptions & {
+    precision_treshold?: number;
+  };
+  percentiles: {
+    field: string;
+    percents?: number[];
+  };
+  extended_stats: {
+    field: string;
+  };
+  top_hits: {
+    from?: number;
+    size?: number;
+    sort?: SortOptions;
+    _source?: SourceOptions;
+  };
+  filter: Record<string, any>;
+  filters: {
+    filters: Record<string, any> | any[];
+  };
+  sampler: {
+    shard_size?: number;
+  };
+  derivative: {
+    buckets_path: BucketsPath;
+  };
+  bucket_script: {
+    buckets_path: BucketsPath;
+    script?: Script;
+  };
+}
+
+type AggregationType = keyof AggregationOptionsByType;
+
+type AggregationOptionsMap = Unionize<
+  {
+    [TAggregationType in AggregationType]: AggregationOptionsByType[TAggregationType];
+  }
+> & { aggs?: AggregationInputMap };
+
+export interface AggregationInputMap {
+  [key: string]: AggregationOptionsMap;
+}
+
+type BucketSubAggregationResponse<
+  TAggregationInputMap extends AggregationInputMap | undefined
+> = TAggregationInputMap extends AggregationInputMap
+  ? AggregationResponseMap<TAggregationInputMap>
+  : {};
+
+interface AggregationResponsePart<
+  TAggregationOptionsMap extends AggregationOptionsMap
+> {
+  terms: {
+    buckets: Array<
+      {
+        doc_count: number;
+        key: string | number;
+      } & BucketSubAggregationResponse<TAggregationOptionsMap['aggs']>
+    >;
+  };
+  histogram: {
+    buckets: Array<
+      {
+        doc_count: number;
+        key: number;
+      } & BucketSubAggregationResponse<TAggregationOptionsMap['aggs']>
+    >;
+  };
+  date_histogram: {
+    buckets: Array<
+      {
+        doc_count: number;
+        key: number;
+        key_as_string: string;
+      } & BucketSubAggregationResponse<TAggregationOptionsMap['aggs']>
+    >;
+  };
+  avg: MetricsAggregationResponsePart;
+  sum: MetricsAggregationResponsePart;
+  max: MetricsAggregationResponsePart;
+  min: MetricsAggregationResponsePart;
+  value_count: MetricsAggregationResponsePart;
+  cardinality: {
+    value: number;
+  };
+  percentiles: {
+    values: Record<string, number>;
+  };
+  extended_stats: {
+    count: number;
+    min: number | null;
+    max: number | null;
+    avg: number | null;
+    sum: number | null;
+    sum_of_squares: number | null;
+    variance: number | null;
+    std_deviation: number | null;
+    std_deviation_bounds: {
+      upper: number | null;
+      lower: number | null;
+    };
+  };
+  top_hits: {
+    hits: {
+      total: {
+        value: number;
+        relation: 'eq' | 'gte';
+      };
+      max_score: number | null;
+      hits: Array<{
+        _source: unknown;
+      }>;
```

Comment:
Is there a reason you're not using the hits type?

Reply:
Just did not considering, that's a great suggestion actually. I'll give it a shot.